olp-cpp-sdk-core/src/client/PendingUrlRequests.cpp

```diff
@@ -33,6 +33,9 @@ olp::client::HttpResponse GetCancelledResponse() {
 namespace olp {
 namespace client {
 
+PendingUrlRequest::PendingUrlRequest(CancellationContext context)
```

Comment:
Not sure this works. Consider this. The first request comes in and creates a PendingUrlRequest with its CancellationContext embedded. Now there is a second and a third request to the same URL that is merged inside the PendingUrlRequest. Now the first request is cancelled, which means we should only remove its callback from the callback list but because we used its CancellationContext as input the entire PendingUrlRequest is cancelled, incl. the second and third request which actually are still waiting for the download. Dont this this is good.

Reply:
This is reverted change